From 5d566765e8e4dc623a17c56f8ae88ecfdd357773 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty <etaash.mathamsetty@gmail.com>
Date: Tue, 22 Jul 2025 15:26:51 -0400
Subject: [PATCH] ntoskrnl.exe: Implement PsGetProcessPeb.

(cherry picked from commit 6aac1bdb1cb2d3ccc9612f6e58fff3596b56d2dc)
---
 include/ddk/ntddk.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/ddk/ntddk.h b/include/ddk/ntddk.h
index 887d1749662..038dc36ae2a 100644
--- a/include/ddk/ntddk.h
+++ b/include/ddk/ntddk.h
@@ -289,6 +289,7 @@ NTSTATUS  WINAPI KeExpandKernelStackAndCallout(PEXPAND_STACK_CALLOUT,void*,SIZE_
 void      WINAPI KeSetTargetProcessorDpc(PRKDPC,CCHAR);
 BOOLEAN   WINAPI MmIsAddressValid(void *);
 HANDLE    WINAPI PsGetProcessId(PEPROCESS);
+PEB *     WINAPI PsGetProcessPeb(PEPROCESS);
 void *    WINAPI PsGetProcessSectionBaseAddress(PEPROCESS);
 HANDLE    WINAPI PsGetThreadId(PETHREAD);
 HANDLE    WINAPI PsGetThreadProcessId(PETHREAD);
